include/envoy/network/connection.h

```diff
@@ -120,6 +120,7 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * @return The address of the remote client.
    */
   virtual const Address::Instance& remoteAddress() PURE;
+  virtual Address::InstanceConstSharedPtr remoteAddressConstSharedPtr() PURE;
```

Comment:
Right, I understand now. (See other comment thread). I think we should figure out a better solution here per the other comment thread.